packages/playground/cli/src/setup-php.ts

```diff
@@ -1,23 +1,189 @@
+import fs from 'fs';
 import { NodePHP } from '@php-wasm/node';
-import { rotatePHPRuntime } from '@php-wasm/universal';
+import {
+	BasePHP,
+	PHPRequestHandler,
+	SupportedPHPVersion,
+	__private__dont__use,
+	rotatePHPRuntime,
+} from '@php-wasm/universal';
 import { rootCertificates } from 'tls';
-import { Mount } from './cli';
+import {
+	defineSiteUrl,
+	runWpInstallationWizard,
+	unzip,
+	zipWpContent,
+} from '@wp-playground/blueprints';
+import {
+	CACHE_FOLDER,
+	cachedDownload,
+	getWordPressVersionUrlAndName,
+	readAsFile,
+} from './download';
+import { dirname } from '@php-wasm/util';
+import path from 'path';
+import { EmscriptenDownloadMonitor } from '@php-wasm/progress';
 
-export async function createPhp(mounts: Mount[]) {
+export async function setupWordPress(
+	php: NodePHP,
+	wpVersion = 'latest',
+	monitor: EmscriptenDownloadMonitor
+) {
+	const wpDetails = await getWordPressVersionUrlAndName(wpVersion);
+	const [wpZip, sqliteZip] = await Promise.all([
+		cachedDownload(wpDetails.url, `${wpDetails.version}.zip`, monitor),
+		cachedDownload(
+			'https://github.com/WordPress/sqlite-database-integration/archive/refs/heads/main.zip',
+			'sqlite.zip',
+			monitor
+		),
+	]);
+	await prepareWordPress(php, wpZip, sqliteZip);
+
+	const prebuiltWpContentPath = path.join(
+		CACHE_FOLDER,
+		`prebuilt-wp-content-for-wp-${wpDetails.version}.zip`
+	);
+	if (fs.existsSync(prebuiltWpContentPath)) {
+		await unzip(php, {
+			zipFile: readAsFile(prebuiltWpContentPath),
+			extractToPath: '/wordpress',
+		});
+		return;
+	}
+
+	await installWordPress(php);
+	const wpContent = await zipWpContent(php);
+	fs.writeFileSync(prebuiltWpContentPath, wpContent);
+}
+
+async function prepareWordPress(php: NodePHP, wpZip: File, sqliteZip: File) {
+	await unzip(php, {
+		zipFile: wpZip,
+		extractToPath: '/tmp',
+	});
+	php.mv('/tmp/wordpress', '/wordpress');
```

Comment:
if we're replacing `/wordpress` anyway, why don't we unzip to that location and skip the unpack+mv sequence?

Reply:
`wordpress.zip` has a `wordpress` directory in it. Technically we can extract specific directories using the `stream-compression` package, but that creates interop issues with JS runtimes so I did the lazy thing to sit it out until we offload the setup to the [PHP library](https://github.com/WordPress/blueprints-library/).

Reply:
can we not have `extractToPath: '/'`?
or you are saying that would override everything in `/` instead of simply adding the `/wordpress` subdirectory?